### PR TITLE
Add new filtering operators 

### PR DESCRIPTION
**Pull request by burov**
I add new filtering operators, such as in(val in array for number and string arrays) and := (Insensitive equal operator)

### CODE REVIEW

File: gorm/filtering_test.go [hunk 1]
```diff
@@ -2,11 +2,11 @@ package gorm
 
 import (
 	"context"
+	"github.com/infobloxopen/atlas-app-toolkit/query"
```
Comment (Evgeniy-L) [inline gorm/filtering_test.go]: why the order of imports was changed. Typically we try to break imports in 3 different sections with a newline between each one to separate standard imports, third party imports, imports from infobloxopen

Comment (burov) [reply gorm/filtering_test.go]: Sorry, i think it because i did use standart go format tool and it perform sorting imports as one of format operation, i will try be more careful about this

File: README.md [hunk 2]
```diff
@@ -1004,6 +1004,8 @@ Literal values include numbers (integer and floating-point), and quoted (both si
 | or           | Logical OR               | price <= 3.5 or price > 200                              |
 | not          | Logical NOT              | not price <= 3.5                                         |
 | ()           | Grouping                 | (priority == 1 or city == ‘Santa Clara’) and price > 100 |
+| :=           | Insensitive equal        | city := 'SaNtA ClArA'                                    |
```
[the remark this answers is not in the document]
Comment (burov) [reply README.md]: Evgeniy-L sanded  me documentation and i did all folow the documentation, in this document string representation not exist for ':=' operator and i did't add it, if you think we need alias 'ie' for ':=' operator, i can add it

File: gorm/filtering_test.go [hunk 3]
```diff
@@ -216,6 +216,41 @@ func TestGormFiltering(t *testing.T) {
...
+			nil,
+			nil,
+		},
+		{
+			"(not(field2 in ['sOmeId', 'egegeg']) and (not(field := 'Hello')))",
+			"((entities.field2 NOT IN (?, ?)) AND NOT(lower(field) = lower(?)))",
+			[]interface{}{"sOmeId", "egegeg", "Hello"},
+			nil,
+			nil,
+		},
```
Comment (dkukharau) [inline gorm/filtering_test.go]: Need a test case for `ArrayNumberCondition` as well - not a complex expression, just a simple one.

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (Evgeniy-L) [reply README.md]: @dkukharau, @burov, `ie` doesn't look intuitively clear. Have you seen something like this anywhere it is is just something we just invented?

[the remark this answers is not in the document]
Comment (Evgeniy-L) [reply README.md]: I don't like the idea of adding uncommon things

[the remark this answers is not in the document]
Comment (dkukharau) [reply README.md]: Ok, let's name it `ieq`, there is such operator in PowerShell.

File: query/filtering_lexer_test.go [hunk 4]
```diff
@@ -37,6 +36,9 @@ func TestFilteringLexer(t *testing.T) {
 		LeToken{},
 		LeToken{},
 		NullToken{},
+		InsensitiveEqToken{},
```
Comment (dkukharau) [inline query/filtering_lexer_test.go]: InToken is missing..

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (burov) [reply README.md]: @Evgeniy-L  what you think about this operator, what we need to? Change alias to 'ieq' or remove alias ?

File: query/filtering_lexer_test.go (hunk 4 above)
Comment (dkukharau) [reply query/filtering_lexer_test.go]: Also good to have negative tests for empty array(`[]`), array with both strings and numbers(`["abc", 1, 2]`), unclosed array(`[1,2,`)

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (Evgeniy-L) [reply README.md]: ok, lets have it added
